grafana-loki,promtail: 2.7.4 -> 2.8.0, remove buildGoModule pin, add myself as maintainer

emilylange opened the pull request:
###### Description of changes

https://github.com/grafana/loki/releases/tag/v2.8.0
https://github.com/grafana/loki/releases/tag/v2.7.5
https://github.com/grafana/loki/blob/4ff61d4a1b70c1f64fa7fd367295eaee336cf070/CHANGELOG.md#280-2023-04-04
diff: https://github.com/grafana/loki/compare/v2.7.4...v2.8.0

Also removed the `buildGo119Module` pin which was needed in < v2.7.4 (#218904, #219885)
From https://github.com/grafana/loki/releases/tag/v2.7.4
> Go version upgraded to `1.20.1` 

And added myself as maintainer :)

Somewhat related: #223754

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).


---

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>grafana-loki</li>
    <li>promtail</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2010

SuperSandro2000 commented on pkgs/servers/monitoring/loki/default.nix:
```diff
@@ -54,7 +54,7 @@ buildGoModule rec {
     homepage = "https://grafana.com/oss/loki/";
-    maintainers = with maintainers; [ willibutz globin mmahut ];
+    maintainers = with maintainers; [ willibutz globin mmahut indeednotjames ];
     platforms = platforms.unix;
```
```suggestion
```
already set by buildGoModule

IndeedNotJames replied:
From what I can tell, `lib.platforms.darwin ++ lib.platforms.linux` != `lib.platforms.unix`.
Can you elaborate?

```bash
# platforms.unix
❯ nix eval --json .#grafana-loki.meta.platforms | jq 'length'
41
# unset
❯ nix eval --json .#grafana-loki.meta.platforms | jq 'length'
22
```
https://github.com/NixOS/nixpkgs/blob/b1581c36189e5063163acb24822ed8a343bf6a8f/pkgs/build-support/go/module.nix#L320
https://github.com/NixOS/nixpkgs/blob/b1581c36189e5063163acb24822ed8a343bf6a8f/pkgs/development/compilers/go/1.20.nix#L180

SuperSandro2000 replied:
Go only support darwin and linux and if that ever changes in the future, we likely do not need to update meta.platforms for every package.

IndeedNotJames replied:
Took a decent while to compile from scratch:
```bash
❯ nix-build -A pkgsCross.x86_64-freebsd.grafana-loki
/nix/store/3l3irkk9aw6h9dgx6qn5nmik4bym5g6v-grafana-loki-x86_64-unknown-freebsd13-2.7.4
❯ file result/bin/*
result/bin/logcli:      ELF 64-bit LSB executable, x86-64, version 1 (FreeBSD), dynamically linked, interpreter /libexec/ld-elf.so.1, Go BuildID=6lZq-V9UeCRKTYdXek8f/NYD_WYwJaCF580YhsjNI/9jb1AW3bhPtPnNLcWZuF/PXcUiWiKviscMaEru4wZ, stripped
result/bin/loki:        ELF 64-bit LSB executable, x86-64, version 1 (FreeBSD), dynamically linked, interpreter /libexec/ld-elf.so.1, Go BuildID=tRS_s9fFV4EnYqCD83qt/ZXynWpT4OP_AN_Z5LsWK/Wsy56FglfdPorzZlKkBb/vEJ8cl8xa-LW40dAYBOK, stripped
result/bin/loki-canary: ELF 64-bit LSB executable, x86-64, version 1 (FreeBSD), dynamically linked, interpreter /libexec/ld-elf.so.1, Go BuildID=Bc4Woh_7Wyc4TM3sFUqW/QAat7p3fnb58-WEv_5i6/xq-ITXlM1oRq4KIh1H_6/NeMsLmDRD8THPpAq1USx, stripped
result/bin/promtail:    ELF 64-bit LSB executable, x86-64, version 1 (FreeBSD), dynamically linked, interpreter /libexec/ld-elf.so.1, Go BuildID=v8sMpP1An8VjLW799gya/uM8d1x6OZOSnXdSD30Eb/ApOIDl0t5Ux9gsggToa6/X14UW_THl-9d642WG0g3, stripped
```

Upstream officially supports freebsd as target, and I see no reason why nixpkgs should drop it.

Is anyone actually using nixpkgs to compile loki for freebsd?
Lol no, extremely unlikely.

Should we actively prevent it?
No, I don't think so.

SuperSandro2000 replied:
> Upstream officially supports freebsd as target, and I see no reason why nixpkgs should drop it.

Our go does not support freebsd. If that supports freebsd it will be automatically added to here

IndeedNotJames replied:
I can't quite follow you, but whatever.

Then why don't you do a treewide change?
With this rather primitive query there are seemingly 141 files using `buildGoModule` and `platforms.unix`.
https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+buildGoModule+%22platforms+%3D%22+unix&type=code

K900 commented:
2.8 is out: https://grafana.com/blog/2023/04/06/grafana-loki-2.8-release-tsdb-ga-logql-enhancements-and-a-third-target-for-scalable-mode/

emilylange committed:
grafana-loki,promtail: add indeednotjames as maintainer

emilylange committed:
grafana-loki,promtail: remove unnecessary go 1.19 pin since v2.7.4

Seems to have slipped through #219885 (a119b197e068a518dc6f4d048b9166b520c94b6f)

See https://github.com/grafana/loki/releases/tag/v2.7.4

emilylange committed:
grafana-loki,promtail: 2.7.4 -> 2.8.0

https://github.com/grafana/loki/releases/tag/v2.8.0
https://github.com/grafana/loki/releases/tag/v2.7.5
https://github.com/grafana/loki/blob/4ff61d4a1b70c1f64fa7fd367295eaee336cf070/CHANGELOG.md#280-2023-04-04
diff: https://github.com/grafana/loki/compare/v2.7.4...v2.8.0

SuperSandro2000 commented on pkgs/servers/monitoring/loki/default.nix:
```diff
@@ -8,17 +8,17 @@
   pname = "grafana-loki";
 
   src = fetchFromGitHub {
     rev = "v${version}";
     owner = "grafana";
     repo = "loki";
```
```suggestion
    owner = "grafana";
    repo = "loki";
    rev = "v${version}";
```

emilylange committed:
grafana-loki,promtail: remove redundant `meta.platforms` and `doCheck`

already set by `buildGoModule`
